robotiq_s_model_articulated_gazebo_plugins/src/RobotiqHandPlugin.cpp

```diff
@@ -0,0 +1,877 @@
+/*
+ * Copyright 2014 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <robotiq_s_model_articulated_msgs/SModelRobotInput.h>
+#include <robotiq_s_model_articulated_msgs/SModelRobotOutput.h>
+#include <ros/ros.h>
+#include <string>
+#include <vector>
+#define private public 
```

Comment:
I needed to change the PID.hh file to allow public access of its members.
They're private in 1.9 and getter functions aren't implemented until gazebo
3.0. I thought that was silly, and so I hacked around it. The functions
directly access the private members by necessity. It can only change if
gazebo patches 1.9 and 2.2 to include getters for it.
On Jan 18, 2015 6:18 AM, "G.A. vd. Hoorn" <notifications@github.com> wrote:

> In robotiq_s_model_articulated_gazebo_plugins/src/RobotiqHandPlugin.cpp
> <https://github.com/ros-industrial/robotiq/pull/17#discussion-diff-23131673>
> :
>
> > + *     http://www.apache.org/licenses/LICENSE-2.0
> > + *
> > + * Unless required by applicable law or agreed to in writing, software
> > + * distributed under the License is distributed on an "AS IS" BASIS,
> > + * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > + * See the License for the specific language governing permissions and
> > + * limitations under the License.
> > + *
> > +*/
> > +
> > +#include <robotiq_s_model_articulated_msgs/SModelRobotInput.h>
> > +#include <robotiq_s_model_articulated_msgs/SModelRobotOutput.h>
> > +#include <ros/ros.h>
> > +#include <string>
> > +#include <vector>
> > +#define private public
>
> I cannot find this in
> bitbucket.org/osrf/drcsim/src/drcsim_gazebo_ros_plugins/src/RobotiqHandPlugin.cpp
> <https://bitbucket.org/osrf/drcsim/src/7aee32718c4a74419b2bb447ab41d8ab9e4edf68/drcsim_gazebo_ros_plugins/src/RobotiqHandPlugin.cpp?at=default>:
> could you explain why this is needed?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ros-industrial/robotiq/pull/17/files#r23131673>.
>